xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/internal/junit5/ExtensionInstaller.java

```diff
@@ -218,75 +210,8 @@ private boolean isSupportedExtensionType(String type)
     public void installExtensions(Collection<ExtensionId> extensions, UsernamePasswordCredentials credentials,
         String installUserReference, List<String> namespaces, boolean failOnExist) throws Exception
     {
-        String xwikiRESTURL = String.format("%s/rest", DockerTestUtils.getXWikiURL(this.context));
-
-        // Resolve the extensions versions if needed
-        List<ExtensionId> resolvedExtensions = new ArrayList<>(extensions.size());
-        for (ExtensionId extensionId : extensions) {
-            String version;
-            if (extensionId.getVersion() == null) {
-                // TODO: search the version of the extension in the dependency tree
-                version = this.mavenResolver.getModelFromCurrentPOM().getVersion();
-            } else {
-                version = this.mavenResolver.replacePropertiesFromCurrentPOM(extensionId.getVersion().getValue());
-            }
-
-            resolvedExtensions.add(new ExtensionId(extensionId.getId(), version));
-        }
-
-        // Install the extensions
-        try {
-            installExtensions(xwikiRESTURL, resolvedExtensions, credentials, installUserReference, namespaces,
+        new RestExtensionInstaller(this.componentManager, this.mavenResolver)
```

Comment:
Should the RestExtensionInstaller be a field too instead of creating a new one for each call of installExtensions ?

Reply:
done, RestExtensionInstaller is now initialized in the constructor